### ISSUE TITLE
CalculateInfoHash method returning different value than original torrent file

### ISSUE THREAD

**Issue by carlosjdelgado** [Krusen/BencodeNET#25]
Hi,

I am trying to get the infohash string from this file:
[TestFile.zip](https://github.com/Krusen/BencodeNET/files/1174312/TestFile.zip)

I know that correct info hash of this file is 037e4ddf1e84fe4d4290fd40c442af2201750849 but when I call to TorrentUtil.CalculateInfoHash this returns 42DCB1CBA3ADD25D1498D2FA8F84DA46EC944BFB that is an incorrect value.

This is an example of code I am using:
```
var bencodeParser = new BencodeParser();
var stream = File.OpenRead("Test.torrent");
var torrentFileData = bencodeParser.Parse<Torrent>(stream);
var infoHash = TorrentUtil.CalculateInfoHash(torrentFileData); 
//Expected value is 037e4ddf1e84fe4d4290fd40c442af2201750849
```

I am using bencodeNET over netcoreapp1.1

**Comment by Krusen** [comment]
The issue is that your torrent contains extra fields for each file (`crc32`, `md5`, `sha1`, `mtime`) which are not in the spec. 

The `Torrent` class was mostly meant for easy access to and manipulation of torrent details while following the spec. It's currently handling extra fields at the root and in `info` but not deeper than that at the moment. 

I might add support for that or add a property like `torrent.OriginalInfoHash`. For now you can work around this issue by parsing as `BDictionary` instead of `Torrent`.

```csharp
var bencodeParser = new BencodeParser();
var torrent = bencodeParser.Parse<BDictionary>("Test.torrent");
var info = torrent.Get<BDictionary>("info");
var infoHash = TorrentUtil.CalculateInfoHash(info);
```

**Comment by Krusen** [comment]
I've released version 2.2.3 on NuGet.

You can now do this:

```csharp
var bencodeParser = new BencodeParser();
var torrent = bencodeParser.Parse<Torrent>("Test.torrent");
var infoHash = torrent.OriginalInfoHash;
var infoHashBytes = torrent.OriginalInfoHashBytes;
```

At some point I might add support for extra fields in the `files` element.
